crypto/fipsmodule/kbkdf/kbkdf_test.cc

```diff
@@ -0,0 +1,641 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/digest.h>
+#include <openssl/err.h>
+#include <openssl/kbkdf.h>
+
+#include <gtest/gtest.h>
+#include "../../test/test_util.h"
+
+// Test vectors from NIST CAVP for SP 800-108
+// https://csrc.nist.gov/Projects/cryptographic-algorithm-validation-program/key-derivation
```

Comment:
Overall A+ testing here, I like the paramaetrized tests and good coverage. However, I'd like to simplify these a bit: our `FileTest` can parse and read the files as is from NIST. This makes it easier to review (make sure there were no types or conversion errors) and makes this test file smaller. `aead_test.cc` is a good example of testing different parameters with different files/algorithms.

Reply:
I'll look into that, thanks. Without knowing anything about it yet, I'd be concerned that the test vectors from the CAVP page have way too much data in them... if `FileTest` can filter to only run certain vectors that'll be fine, if not, the majority of the CAVP data isn't going to test properly. The spec for KBKDF is stupidly flexible for some reason.

Reply:
If it's OK with you, I'll do this as a separate task, and also create one to adapt the SSHKDF tests to use this... they also have CAVP test data avilable but don't currently use it.

Reply:
CryptoAlg-1436 created for KBKDF